Sources/Signal+SwiftConcurrency.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  Signal+SwiftConcurrency.swift
+//  ReactiveSwift
+//
+//  Created by Marco Cancellieri on 2021-11-11.
+//  Copyright (c) 2021 GitHub. All rights reserved.
+//
+#if compiler(>=5.5.2) && canImport(_Concurrency)
+import Foundation
+
+@available(macOS 10.15, iOS 13, watchOS 6, tvOS 13, macCatalyst 13, *)
+extension Signal {
+	public var asyncThrowingStream: AsyncThrowingStream<Value, Swift.Error> {
+		AsyncThrowingStream<Value, Swift.Error> { continuation in
+			let disposable = observe { event in
+				switch event {
+				case .value(let value):
+					continuation.yield(value)
+				case .completed, .interrupted:
```

Comment:
Should the `interrupted` event throw a swift `CancellationError`?

Reply:
Good point